contrib/packs/actions/workflows/install.yaml

```diff
@@ -1,32 +1,101 @@
----
-  chain:
-    -
-      name: "download pack"
-      ref: "packs.download"
-      parameters:
-        packs: "{{packs}}"
-        force: "{{force}}"
-        python3: "{{python3}}"
-      on-success: "make a prerun"
-    -
-      name: "make a prerun"
-      ref: "packs.virtualenv_prerun"
-      parameters:
-        packs_status: "{{ __results['download pack'].result }}"
-      on-success: "install pack dependencies"
-    -
-      name: "install pack dependencies"
-      ref: "packs.setup_virtualenv"
-      parameters:
-        packs: "{{ __results['make a prerun'].result }}"
-        env: "{{env}}"
-        python3: "{{python3}}"
-      on-success: "register pack"
-    -
-      name: "register pack"
-      ref: "packs.load"
-      parameters:
-        register: "{{register}}"
-        packs: "{{ __results['make a prerun'].result }}"
-
-  default: "download pack"
+version: 1.0
+
+description: A orquesta workflow to install packs.
+
+input:
+  - packs
+  - register
+  - env
+  - force
+  - python3
+  - skip
+
+vars:
+  - packs_list: null
+  - dependency_list: null
+  - conflict_list: null
+  - nested: 3
+
+tasks:
+  init_task:
+    action: core.noop
+    next:
+      - do: download_pack
+
+  download_pack:
+    action: packs.download
+    input:
+      packs: <% ctx().packs %>
+      force: <% ctx().force %>
+      python3: <% ctx().python3 %>
+      dependency_list: <% ctx().dependency_list %>
+    next:
+      - when: <% succeeded() %>
+        do: make_a_prerun
+
+  make_a_prerun:
+    action: packs.virtualenv_prerun
+    input:
+      packs_status: <% task(download_pack).result.result %>
+      packs_list: <% ctx().packs_list %>
+    next:
+      - when: <% succeeded() %>
+        publish:
+          - packs_list: <% task(make_a_prerun).result.result %>
+        do: find_next_action
+
+  find_next_action:
+    action: core.noop
+    next:
+      - when: <% ctx().skip or ctx().nested = 0 %>
+        do: install_pack_requirements
+      - when: <% ctx().nested > 0 and not ctx().skip %>
+        do: get_pack_dependencies
+
+  get_pack_dependencies:
+    action: packs.get_pack_dependencies
+    input:
+      packs_status: <% task(download_pack).result.result %>
+      nested: <% ctx().nested%>
+    next:
+      - when: <% succeeded() %>
+        publish:
+          - dependency_list: <% task(get_pack_dependencies).result.result.dependency_list %>
+          - conflict_list: <% task(get_pack_dependencies).result.result.conflict_list %>
+          - nested: <% task(get_pack_dependencies).result.result.nested %>
+        do: check_dependency_and_conflict_list
+
+  check_dependency_and_conflict_list:
+    action: core.noop
+    next:
+      - when: <% ctx().conflict_list %>
+        do: stop_installation_and_cleanup
+      - when:  <% not ctx().conflict_list and ctx().dependency_list %>
+        do: download_pack
```

Comment:
What happens if a dependency is already downloaded and installed on the system?

Reply:
@m4dcoder If a dependency is already installed.  The exciting pack version will be checked. If version is same, do thing for the dependency, otherwise, failed workflow as conflicting.